DOCSP 24409 Create Diagrams page (#81)
mongodb/docs-relational-migrator: source/diagrams/manage-diagrams/create-diagrams.txt

.. _rm-create-diagrams:

===============
Create Diagrams
===============

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

> You can create new diagrams in the Mapping tab of Relational Migrator.

Reviewer: We should also specify here that you can create multiple diagrams and can select between them at any one time and a slight description of why you would want to do that (viewing specific entities of the data model). Not sure if there is a maximum number of diagrams but it might be good to specify in TR.

> You can create new diagrams in the Mapping tab of Relational Migrator.

Author: Done!

> You can create new diagrams in the Mapping tab of Relational Migrator.

Author: Done!

> You can create new diagrams in the Mapping tab of Relational Migrator.

Author: Confirmed with Lavender that there is no maximum number of diagrams!

After you create a diagram, you can clone, rename, and delete your new
diagrams. 

> 

Author: Done!

To create a new diagram, click the plus sign button on the right of the
workspace tab. This creates and opens a blank diagram. By default, the
diagram view pane hides all entities in the project. 

.. image:: /img/diagrams/create_diagram.png
   :alt: Create Diagram in Diagram View Pane

> When you create a new diagram, the new diagram is the active diagram
> that Relational Migrator displays. 

Reviewer: ```suggestion
When you create a new diagram, it displays on the mapping tab.
```

Author: Done!

.. note::

   Diagram tabs appear in the order that the diagrams were created. You

>    cannot reorder diagram tabs.

Author: Done!

   :depth: 1
   :class: singlecol

You can create new diagrams on the Mapping tab of Relational Migrator.
You can also create multiple diagrams and navigate between them to view
specific entities of your data model. By default, Relational Migrator

> creates the ``Main Diagram`` diagram with a Relational view and a

Reviewer: We should use the `:guilabel:` directive when referring directly to a UI element.
```suggestion
creates the :guilabel:`Main Diagram` with a Relational view and a
```

Author: Done!

MongoDB view.

About this Task
---------------

> Creating diagrams allows you to visualize tables, foreign key

Reviewer: **Suggestion**

Convert these items under AtT to bullet points.

Author: Done! Thanks for your feedback on this one!

> relationships, and data types in different formats. After you create a

Reviewer: ```suggestion
relationships, and data types. After you create a
```

diagram, you can :ref:`clone <rm-clone-diagrams>`, :ref:`rename
<rm-rename-diagrams>`, and :ref:`delete <rm-delete-diagrams>` your new
diagram. 

You must have at least one digram on your ribbon tab. 

Steps
-----

.. procedure::
   :style:  connected

>    .. step:: Click the plus sign button on the right of the workspace tab 

Reviewer: **Suggestion**

I like to use the icon role when referring to UI elements such as these, it can help with non-english readers. We also have the ability to match the exact leafy green icons that this app was built with, I put [an edit into the style guide which lists out all available icons ](https://www.mongodb.com/docs/meta/reference/icon/#rendered-icon-table):). 

```suggestion
   .. step:: Click the :icon-lg:`Plus` button on the right of the workspace tab 
```

Author: Done!

>       .. image:: /img/diagrams/create_diagram.png

Reviewer: **Suggestion**

I like the image here to center the user in the UI. Could you add a red square highlight on + button just for extra clarity? 

Something like:

![Screenshot 2024-01-10 at 1 41 47 PM](https://github.com/mongodb/docs-relational-migrator/assets/85948430/a747f99b-775b-4186-8253-1e902ab9449c)

Author: Done!

   :depth: 1
   :class: singlecol

You can create new diagrams on the :guilabel:`Mapping` screen of
Relational Migrator. You can also create multiple diagrams and navigate
between them to view specific entities of your data model. By default,
Relational Migrator creates the :guilabel:`Main Diagram` with a
Relational view and a MongoDB view.

About this Task
---------------

- Creating diagrams allows you to visualize tables, foreign key
  relationships, and data types. 
- After you create a diagram, you can
  :ref:`clone <rm-clone-diagrams>`, :ref:`rename <rm-rename-diagrams>`,
  and :ref:`delete <rm-delete-diagrams>` your new diagram. 
- You must have at least one digram on your Diagram Toolbar. 
- Relational Migrator includes diagrams when you import and export
  projects. 

Steps
-----

.. procedure::
   :style:  connected

   .. step:: Click the :icon-lg:`Plus` button on the Diagram Toolbar
   
      .. image:: /img/diagrams/create-diagram.png
         :alt: Create Diagram in Diagram View Pane

>       This creates and opens a blank diagram. By default, the diagram

Reviewer 2: [note, minimalism] Just a non-blocking note -- let's try to stay away from the "This does x" construction in the future. Instead, consider "X happens."
